### PR TITLE
Move documentation about avoiding bans into a topic of its own

### PR DESCRIPTION
**Pull request by Gallaecio**

### CODE REVIEW

File: docs/topics/avoiding-bans.rst [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+guidelines, set :setting:`ROBOTSTXT_OBEY` to ``True`` to enable the
+:class:`~scrapy.downloadermiddlewares.robotstxt.RobotsTxtMiddleware`
+middleware. When you do, if your spider attempts to crawl a restricted page,
+this middleware aborts that request with the following message::
+
+    Forbidden by robots.txt
+
+Also set :setting:`CONCURRENT_REQUESTS_PER_DOMAIN` and
+:setting:`DOWNLOAD_DELAY` to values that comply with the ``Crawl-Delay`` or
+``Request-Rate`` directives from the ``robots.txt`` guidelines.
```
Comment (kmike) [inline docs/topics/avoiding-bans.rst]: TODO: do it automatically, https://github.com/scrapy/scrapy/issues/892

**Comment by Gallaecio** [comment]
To-Do: cover https://datasetsearch.research.google.com/

File: docs/topics/avoiding-bans.rst [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+Being polite
+============
+
+To avoid being banned, you should first avoid giving a website reasons to ban
+you.
+
+.. _identifying-yourself:
+
+Idenitfying yourself
```
Comment (wRAR) [inline docs/topics/avoiding-bans.rst:43]: Typo

File: docs/topics/avoiding-bans.rst [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+--------------------
+
+If your crawling has a noticeable nevative impact on a website or you crawl
```
Comment (wRAR) [inline docs/topics/avoiding-bans.rst:46]: Typo, "nevative"

File: docs/topics/avoiding-bans.rst [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+
+Also set :setting:`CONCURRENT_REQUESTS_PER_DOMAIN` and
+:setting:`DOWNLOAD_DELAY` to values that comply with the ``Crawl-Delay`` or
+``Request-Rate`` directives from the ``robots.txt`` guidelines.
+
+You may also use the :ref:`AutoThrottle extension <topics-autothrottle>` on top
+of that, so that when the target website experiences a high load, your spider
+automatically switches to higher download delays.
+
+.. _robot.txt Google specification: https://developers.google.com/search/reference/robots_txt
```
Comment (wRAR) [inline docs/topics/avoiding-bans.rst:84]: Typo, "robot.txt" (same where is referenced)

File: docs/topics/avoiding-bans.rst [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+Whichever solution you choose, implement it as a :ref:`downloader middleware
+<topics-downloader-middleware>` that automatically detects CAPTCHA challenges
+in responses and solves them, so that your spider code only receives successful
+responses.
+
+.. _CAPTCHA challenge: https://en.wikipedia.org/wiki/CAPTCHA
+.. _optical character recognition: https://en.wikipedia.org/wiki/Optical_character_recognition
+.. _pytesseract: https://github.com/madmaze/pytesseract
+.. _scrapy-crawlera: https://scrapy-crawlera.readthedocs.io/en/latest/
```
Comment (wRAR) [inline docs/topics/avoiding-bans.rst:221]: This can be moved to the block where it's mentioned

File: docs/topics/avoiding-bans.rst [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+
+An increasing number of websites use solutions that apply many of the above
+filters and challenges at the same time.
+
+There are paid proxy services, like `Zyte Smart Proxy Manager`_, that
+automatically bypass website filters and challenges, so that your spider only
+gets successful responses. They also allow managing sessions to simulating user
+behavior.
+
+For Zyte Smart Proxy Manager, installing scrapy-crawlera_ will ofter advanced
```
Comment (wRAR) [inline docs/topics/avoiding-bans.rst:253]: Typo, "ofter"

**Review by wRAR — APPROVED** [review]
Approved with trivial suggestions

**Comment by Gallaecio** [comment]
I was thinking of covering `DOWNLOADER_CLIENT_TLS_METHOD` as well now, but I don’t want to delay this further, I’ll create a separate ticket for that.

**Pull request merged**